streams/src/main/java/org/apache/kafka/streams/state/internals/FilteredCacheIterator.java

```diff
@@ -24,11 +24,48 @@
 class FilteredCacheIterator implements PeekingKeyValueIterator<Bytes, LRUCacheEntry> {
     private final PeekingKeyValueIterator<Bytes, LRUCacheEntry> cacheIterator;
     private final HasNextCondition hasNextCondition;
+    private final PeekingKeyValueIterator<Bytes, LRUCacheEntry> wrappedIterator;
 
     FilteredCacheIterator(final PeekingKeyValueIterator<Bytes, LRUCacheEntry> cacheIterator,
-                          final HasNextCondition hasNextCondition) {
+                          final HasNextCondition hasNextCondition,
+                          final CacheFunction cacheFunction) {
         this.cacheIterator = cacheIterator;
         this.hasNextCondition = hasNextCondition;
+        this.wrappedIterator = new PeekingKeyValueIterator<Bytes, LRUCacheEntry>() {
```

Comment:
Could we not pass this iterator in via the ctor? Wouldn't need the `CacheFunction` anymore and would mean that we could write tests just for this if need be